### PR TITLE
feat: Adds macOS plugin

### PR DESCRIPTION
**Pull request by StefanKurek**
### Proposed Change
Adds the macOS plugin

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: plugins/macos_logs.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+version: 0.0.1
+title: MacOS
```
Comment (cpheps) [inline plugins/macos_logs.yaml:2]: should probably be `macOS`

File: plugins/macos_logs.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+description: Log parser for MacOS
```
Comment (cpheps) [inline plugins/macos_logs.yaml:3]: Should be `macOS`

File: plugins/macos_logs.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            parse_from: attributes.timestamp
+            layout_type: gotime
+            layout: '2006-01-02 15:04:05-07'
+        - id: time_parser_without_timezone
+          type: regex_parser
+          regex: '^(?P<timestamp>\w{3}\s*\d{1,2} \d{2}:\d{2}:\d{2}) (---|(?P<host>[^ ]*))? ((?P<process>[^\[]*)\[(?P<pid>[^\]]*)\])?( \((?P<subprocess>[^\[]*)(\[(?P<spid>[^\]]*)\])?\))?(: )?(?P<message>[\w\W]*)'
+          timestamp:
+            parse_from: attributes.timestamp
+            layout_type: gotime
+            layout: 'Jan _2 15:04:05'
```
Comment (cpheps) [inline plugins/macos_logs.yaml:79]: I think you need and add processor here just to point the two timestamp blocks back to a single point. See [mysql](https://github.com/observIQ/observiq-otel-collector/blob/main/plugins/mysql_logs.yaml#L634-L637) as example.

Comment (StefanKurek) [reply plugins/macos_logs.yaml:78]: fixed

File: plugins/macos_logs.yaml (hunk 2 above)
Comment (StefanKurek) [reply plugins/macos_logs.yaml]: fixed

File: plugins/macos_logs.yaml (hunk 1 above)
Comment (StefanKurek) [reply plugins/macos_logs.yaml]: fixed

**Review by cpheps — APPROVED** [review]

**Review by BinaryFissionGames — APPROVED** [review]

**Pull request merged**